src/EnergyPlus/Material.hh

```diff
@@ -201,35 +205,50 @@ namespace Material {
         Real64 SpecHeat = 0.0;  // Layer specific heat (J/kgK)
         Real64 Thickness = 0.0; // Layer thickness (m)
 
-        virtual ~MaterialBase()
-        {
-        }
+        Real64 AbsorpThermal = 0.0;              // Layer thermal absorptance
```

Comment:
Added some fields to `MaterialBase` so that we don't have to `dynamic_cast<MaterialChild *>` quite so often. FWIW a `dynamic_cast` is a relatively cheap operation at runtime. It's effectively a check of whether the virtual function table pointer of the object is equal to the virtual function table pointer of the class the object is being cast to (this is why a class needs to have at least one virtual function, by default a destructor, if you want to be able to `dynamic_cast` down to it), where `this` is set to `nullptr` if the check fails. The problem is code clutter more than performance.